packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -30,8 +30,8 @@ export type ImageProps = Omit<
 > & {
   src: string | StaticImport
   alt: string
-  width?: number | `${number}`
-  height?: number | `${number}`
+  width: number | `${number}`
+  height: number | `${number}`
```

Comment:
How about the the rest of the props in this file such as `fill` and `loader`? Please explain why only width and height are causing issues when declared as optional.

Reply:
- Made `width` and `height` mandatory when `fill` is not set or is `false`.
- Allowed `width` and `height` to be optional when `fill` is `true`.
- Improved type safety for ImageProps to ensure consistent usage in components.